bdk_keychain/src/keychain_txout_index.rs

```diff
@@ -257,6 +266,28 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
                 .unwrap()
         }
     }
+
+    pub fn trailing_unused_count(&self, keychain: &K) -> usize {
+        let last_index = self
+            .inner
+            .range_unused((keychain.clone(), u32::MIN)..=(keychain.clone(), u32::MAX))
+            .last()
+            .map(|(&(_, index), _)| index)
+            .unwrap_or(0);
```

Comment:
Related to below. I think it's more useful and intuitive to have provide `last_active_index(&self, keychain: &K) -> Option<u32>`. Trailing unused is `derivation_index` - `last_active_index`.